keras/backend.py

```diff
@@ -4597,10 +4612,12 @@ def _get_input_tensor(time):
     # the value is discarded.
     output_time_zero, _ = step_function(
         input_time_zero, tuple(initial_states) + tuple(constants))
+
+    output_ta_size = time_steps_t if return_all_outputs else 1
```

Comment:
When `return_all_outputs=False` we don't need the TensorArray logic at all, right?

Reply:
Totally agree @fchollet . I just wanted to change the logic as little as possible. By having a common TensorArray, the only things that change when `return_all_outputs=False` is the size and that we always write at index 0; similar to what I did in the `unroll=True` branch when keeping a list of one element.

Of course, suggestions are more than welcome to improve this :)